src/nvim/getchar.c

```diff
@@ -912,9 +913,10 @@ int ins_typebuf(char_u *str, int noremap, int offset, int nottyped, bool silent)
     memmove(s1 + newoff + offset, str, (size_t)addlen);
     /* copy the old chars, after the insertion point, including the	NUL at
      * the end */
+    size_t bytes;
+    STRICT_SUB(typebuf.tb_len, offset + 1, &bytes, size_t);
```

Comment:
looks correct to me.

`typebuf.tb_len - ( offset + 1) ==  (typebuf.tb_len +1) - offset`

Reply:
no, it's the same problem as https://github.com/neovim/neovim/commit/78f36fcf8c37bd51d81e9cf17de588a704a29769

Reply:
@justinmk would this be a correct solution?
` STRICT_SUB(typebuf.tb_len +1, offset, &bytes, size_t);`

I don't understand how this is causing problems.

Reply:
> I don't understand how this is causing problems.

I didn't look at this case specifically, but in general as @erw7 said in https://github.com/neovim/neovim/pull/10508#discussion_r303277429 , it's because of grouping.  `(A - B + C)` is not necessarily the same as `(A - (B + C))`. Moving `+1` to the first group (i.e. `typebuf.tb_len + 1`) avoids the problem.